Create playerdata.js

aldldl opened the pull request:
Create the file for playerdata command. Will update file shortly

aldldl committed:
Create playerdata.js

Create the file for playerdata command. Will update file shortly

aldldl committed:
Test player data

First test of player data

aldldl committed:
Update playerdata.js

$name = ${name}

aldldl committed:
Update playerdata.js

`${name}` is not the same as ${name}

aldldl committed:
added PlayerData json call

aldldl committed:
Update playerdata.js

moved playerdata

aldldl committed:
Update playerdata.js

.

aldldl committed:
Update playerdata.js

test with brackets

aldldl committed:
1

aldldl committed:
spell console

aldldl committed:
2

aldldl committed:
Update playerdata.js

aldldl committed:
2

aldldl committed:
2

aldldl committed:
3

3

aldldl committed:
3

3

aldldl committed:
typo1

fixed a typo

aldldl committed:
player data typo

= > does not equal =>

aldldl committed:
Update playerdata.js

typo 3

aldldl committed:
try-catch

test try-catch

aldldl committed:
fixed finaldata spaces

spaces

aldldl committed:
try catch, with var

test with let

aldldl committed:
typeof-check

added undefined check v1

aldldl committed:
added new check for name

added new check for name

aldldl committed:
added err

add err testcase

aldldl committed:
changed to Error

aldldl committed:
check final data again

check with .error

aldldl committed:
order of catch

aldldl committed:
fix catch

fix catch?

aldldl committed:
errors again

errors

aldldl committed:
change to msg send

msg send

aldldl committed:
Update playerdata.js

change msg to catch

aldldl committed:
Update playerdata.js

remove an extra }

aldldl committed:
mydatafix-tov

fixed to do if mydata

aldldl committed:
add msg send

added msg send

aldldl committed:
stringify

stringify

aldldl committed:
Update playerdata.js

formatting

aldldl committed:
Update playerdata.js

test regex

aldldl committed:
Update playerdata.js

regex

aldldl committed:
Update playerdata.js

again

aldldl committed:
Update playerdata.js

aldldl committed:
Update playerdata.js

aldldl committed:
Update playerdata.js

revert back to odd formating

aldldl committed:
Update playerdata.js

removed player data needs board role

aldldl committed:
permsupdate1

made command perms role || displayName

aldldl committed:
Update playerdata.js

awat

aldldl committed:
Update playerdata.js

added in role by id

aldldl committed:
Update playerdata.js

specified role?

aldldl committed:
Update playerdata.js

removed quotes from role to see if role is role

aldldl committed:
Update playerdata.js

change to 2 step

aldldl committed:
Update playerdata.js

darn semi - typo

aldldl committed:
Update playerdata.js

aldldl committed:
roll back perms

kill changes for perms test 1

aldldl committed:
Update infoBot.js

test out exception roles

aldldl committed:
Update playerdata.js

aldldl committed:
Update infoBot.js

add in quotes

aldldl committed:
Update infoBot.js

removed old code

aldldl committed:
Update infoBot.js

Added additional error details for player data role issue errors.

aldldl committed:
Update playerdata.js

removed extra space behind json, Added better erro when Name not found. Added Console Log when Name not found

aldldl committed:
Create playerdata2

Created playerdata2, I will use this as the new playerdata once I get formatting from PHIDAS correct

Commit:
Formatted the playerdata2

Commit:
update typo

message->msg

Commit:
requrie embed - update playerdata2

Requrie Discord.js for embed

Commit:
Added exception for playerdata2 info

Commit:
added exception

Added the exception for permissions, not just error message

Commit:
Finish Draft of Data2

Use displayName instead of username

Commit:
bug fix

fixed perms for og datastore

Commit:
if no name then use your own - test

Commit:
playerdata2

remove args requirement

Commit:
Try with function

Commit:
fixed catch error

commented out 117 (empty send)

Commit:
added one more exception to roles

Commit:
Added logging features

Added console log when command completes.

Commit:
typo

Commit:
Clean up and usage correction

Commit:
typo?

Commit:
Added helplevel override

Commit:
all override round2

bug fix for help with "all" override

Commit:
change from backtick and single quote to double quote for consistancy

Commit:
New label

Commit:
Updated Server login offline msg

Commit:
dont for get to ecs your chars

Commit:
Not sure what happened

Commit:
no idea why im getting error now 1

Commit:
Edited notice again

Commit:
mis-spelling fix on main index

Commit:
Cleaned up pd and pd2

Cleaned up pd2 and added comments, removed dupes and some commented test code.
Slightly cleaned up pd1, removed unused 'let'(s) and added in a comment about the need for the name in the comment code so people know it is not an oversite.

aldldl commented:
crap is it master or dev I want to use. I think its master because Dev still has Tovs log stuff in it

tovernaar123 commented:
if you want it to run you shood master

tovernaar123 commented on commands/help.js:
```diff
@@ -27,6 +27,10 @@ module.exports = {
 
             data.push(`\`${commandList}\``);
+            //Semi Restricted Public Items//
+                data.push(`**Special (semi-public) Commands:**`);
+                let semiPublic = commands.filter(command => command.helpLevel === "all").map(command => command.name).join(', ');
+                data.push(`\`${semiPublic}\``);
```
not block but the indet is wierd

aldldl replied:
Not sure what you are saying, maybe I should reduce the indent here?

tovernaar123 replied:
was probibly using mobile :)

tovernaar123 commented on commands/playerdata.js:
```diff
@@ -0,0 +1,54 @@
+//player data storage access
+
+module.exports = {
+    name: 'playerdata',
```
mabye rename to playerdataJson and have the other one be playerdata

aldldl replied:
I could see that.

tovernaar123 commented on commands/playerdata.js:
```diff
@@ -0,0 +1,54 @@
+    guildOnly: true,
+    args: true,
+    helpLevel: 'staff',
+    required_role: role.board,
```
we shood allow this to be a function ngl

aldldl replied:
I should know what that means, but I dont...

tovernaar123 replied:
a function is a piece of code that has argument (like exucte) and then that function will rueturn true if the command can be run false if cant

tovernaar123 commented on commands/playerdata.js:
```diff
@@ -0,0 +1,54 @@
+    execute(msg, args, _, internal_error) {
+        let name = args[0];
+                
+        //this command unlike its brother pd2 DOES require a name to be specified. You can still look up yourself but you must add in your username to look up.
```
why?

aldldl replied:
I dont know, I think because I had to jump through alot of hoops for that and I already wrote this one.?.

tovernaar123 replied:
gues its not too bad but it might be nice to allow .exp pd2 with no name

aldldl replied:
Maybe We should create an issue and I'll do that after merge

tovernaar123 commented on commands/playerdata.js:
```diff
@@ -0,0 +1,54 @@
+        }
+        if (name) 
+        {
+            const fs  = require('fs');
```
put requirs at the top of the file above even the export

aldldl replied:
done on line 3 now

tovernaar123 commented on commands/playerdata.js:
```diff
@@ -0,0 +1,54 @@
+    helpLevel: 'staff',
+    required_role: role.board,
+    usage: ` <nameToLookup>`,
+    execute(msg, args, _, internal_error) {
```
make this async (is import for fs.readfile not sync)

aldldl replied:
I copied this from a command you made, Not entirely sure how I can do that, do I just put async in front of it and it magically switches? @tovernaar123

tovernaar123 replied:
yes

aldldl replied:
```suggestion
   async execute(msg, args, _, internal_error) {
```
done?

tovernaar123 commented on commands/playerdata.js:
```diff
@@ -0,0 +1,54 @@
+        if (name) 
+        {
+            const fs  = require('fs');
+            let rawdata = fs.readFileSync('/home/exp_admin/api_v2/persistent_storage.json');
```
if you make the function async you read this with fs.readFile besides that i would really like to see some cache here as this file might become quit big

aldldl replied:
I don't know how cache works.

tovernaar123 commented on commands/playerdata.js:
```diff
@@ -0,0 +1,54 @@
+            const fs  = require('fs');
+            let rawdata = fs.readFileSync('/home/exp_admin/api_v2/persistent_storage.json');
+            let mydata = JSON.parse(rawdata);
+            key1 = `${name}`; // in dataFile
+            let checkdata = mydata["PlayerData"][key1];
```
this can just be 
let checkdata = mydata["PlayerData"][name];

tovernaar123 replied:
yes and then also remove key1 =`${name}`

aldldl replied:
```suggestion
            //key1 = `${name}`; // in dataFile ->-> commented out, will remove after testing. 
            let checkdata = mydata["PlayerData"][name];
```

tovernaar123 commented on commands/playerdata.js:
```diff
@@ -0,0 +1,54 @@
+            let checkdata = mydata["PlayerData"][key1];
+            let finaldata;
+            
+            if(mydata["PlayerData"][key1] )
```
this can be if(checkdata)

aldldl replied:
```suggestion
            if(checkdata)
```
I think I meant that but forgot to change it after I added checkdata above

tovernaar123 replied:
yes

tovernaar123 commented on commands/playerdata.js:
```diff
@@ -0,0 +1,54 @@
+            
+            if(mydata["PlayerData"][key1] )
+            {
+                finaldata = mydata["PlayerData"][key1]["Statistics"];
```
again you have checkdata you can use this here too checkdata["Statistics"]

aldldl replied:
```suggestion
                finaldata = checkdata["Statistics"];
```

tovernaar123 replied:
yes

tovernaar123 commented on commands/playerdata.js:
```diff
@@ -0,0 +1,54 @@
+            if(mydata["PlayerData"][key1] )
+            {
+                finaldata = mydata["PlayerData"][key1]["Statistics"];
+                msg.channel.send(`${name}:\n\`\`\`json\n${JSON.stringify(finaldata, null, 2)}\`\`\``);
```
you need to make sure this cant be more then 2000 chars just to be safe

aldldl replied:
```suggestion
                msg.channel.send(`${name}:\n\`\`\`json\n${JSON.stringify(finaldata, null, 2)}\`\`\``);
```
This should work to split it automatically. Though I will need to test it and then append the ```

tovernaar123 replied:
no as then it wont have the ``` if its more then 2000

aldldl replied:
will have to set up a check later, We should make an issue for this one then.

aldldl replied:
Will have to fix in new issue.

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+    guildOnly: true,
+    args: false,
+    helpLevel: "all",
+    required_role: role.board,
```
again we shood make this allow a function so we can check if name is player name

aldldl replied:
what allow? I thought I added an override to this file? Now I have to go see what I did.

tovernaar123 replied:
no i mean we shood make it possible to add a function as role so that its excutes code to check if somoen shood be able to run the command

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+//required_role has an exception for the user to look upthemselves, nothing needed here it is handeled in infoBot.js...
+//this helpLevel:"all" is required to show up on "semi public commands" it is not needed if the regular command was not restricted to role.board
+module.exports = {
+    name: 'playerdata2',
```
if you rename the other one this can just be playerdata

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                            return 0;
+                        } else {
+                            try {
+                                return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
```
could you add a comment on what this does?

PHIDIAS0303 replied:
thousand seperator

tovernaar123 replied:
what does that mean?

aldldl replied:
1,000
1000
1.000
2 have thousands separated, one like American, one not at all, one like euro.

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+        if(!name){name = msg.member.displayName; playerdata2command(name,msg,args)}
+        
+         
+        function playerdata2command(name,msg,args)
```
this function shood not be inside the excute function

aldldl replied:
I believe you, but why?

tovernaar123 replied:
every time excute runs it waist recourses creating a function that does not use any arguments from the stack above it

aldldl replied:
Will create issue and do it after review is set

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+         
+        function playerdata2command(name,msg,args)
+        {
+            if (1 == 1) 
```
this is dumb

aldldl replied:
This used to be a needed test, for some reason I did not want to lose the if statement so I made something that would always be true. Again I really should have commented that one because I don't remember why. Maybe because Phidas' code had it there and I tried to make it as plug and play as possible.

tovernaar123 replied:
this wont do anything just run the code, so please remove it to make it easier to read :)

aldldl replied:
ok

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                { // start PHIDAS format feature
+
+                    
+                    function ts(x) {
```
what does ts stand for?

PHIDIAS0303 replied:
thousand seperator

aldldl replied:
```suggestion
                    function ts(x) { //thousands separator
```

tovernaar123 replied:
yes thats an good suggestion ^ also sorry for not seeing that

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                            try {
+                                return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+                            } catch(e) {
+                                return 0;
```
make sure this return internal error of the error not just 0

aldldl replied:
I could be wrong but I think you want it as a number because it does math stuffs and *could* be 0 while still wanting the other math stuffs. we should ask @PHIDIAS0303

tovernaar123 replied:
first of all this seems to really not need an try catch just make sure that if the function will error (like x not being a string) something like that that then it just returns 0

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                            'MapTagsMade':'Map Tags Created'
+                        };
+                        
+                        var msg_4 = [];
```
2 things never use var as it hoisted second, what does the name msg_4 mean their is only 1 message being sned right?

aldldl replied:
```suggestion
                        let msg_4 = [];
```

tovernaar123 replied:
does not change why we have msg_4 what does msg_4 mean (but yes  the change of var to let is good)

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                            var msg_3 = [];
+                            
+                            if (layout[i] == 'Playtime' || layout[i] == 'AfkTime') {
+                                try {
+                                    var t1 = msg_2[layout[i]];
+                                } catch (e) {
+                                    var t1 = 0;
+                                }
+                                    
```
Again use let. Second check with an if if layout[i] exists if it does not set t1 to
Third what do these variable names mean.

aldldl replied:
```suggestion
                                try {
                                    let t1 = msg_2[layout[i]];
                                } catch (e) {
                                    let t1 = 0;
                                }
                                    
```
We need to ask @PHIDIAS0303 for more details on this one too

tovernaar123 replied:
yes ^but the change to let is good

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                        var msg_4 = [];
+                    
+                        for (let i = 0; i < layout.length; i++) {
+                            var msg_3 = [];
```
Use let, what is msg_3?

aldldl replied:
```suggestion
                            let msg_3 = [];
```

tovernaar123 replied:
good but still want to know what msg_3 is

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                        for (let i = 0; i < layout.length; i++) {
+                            var msg_3 = [];
+                            
+                            if (layout[i] == 'Playtime' || layout[i] == 'AfkTime') {
```
add some comments on what this is checking as it is confusing me :) .

PHIDIAS0303 replied:
only the value of playtime and afktime need to do some calculation

aldldl replied:
```suggestion
                            if (layout[i] == 'Playtime' || layout[i] == 'AfkTime') { // Playtime and AfkTime are calculated, all others is taken as stored.
```

tovernaar123 replied:
yes okey good ^

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                                    var t1 = 0;
+                                }
+                                    
+                                var h1 = Math.floor(t1 / 60);
+                                var m1 = Math.floor(t1 % 60);
```
what is this doing?

PHIDIAS0303 replied:
minutes to hours and minutes

aldldl replied:
```suggestion
                                let h1 = Math.floor(t1 / 60); //the hours part of min-> hh:mm
                                let m1 = Math.floor(t1 % 60); // the min part of hh:mm
```

tovernaar123 replied:
als use let, second many rename h1 to hours and m1 to minutes

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                                var h1 = Math.floor(t1 / 60);
+                                var m1 = Math.floor(t1 % 60);
+                    
+                                if (h1 > 0) {
+                                    msg_3.push(layout_dict[layout[i]]);
+                                    msg_3.push(h1 + ' h ' + m1 + ' m');
+                                } else {
+                                    msg_3.push(layout_dict[layout[i]]);
+                                    msg_3.push(m1 + ' m ');
+                                }
```
their needs to tell what h1 is and why it could be bigger the 0 and why we check this.

PHIDIAS0303 replied:
we only need to show minutes when it is less than 1 hour of total time

aldldl replied:
```suggestion
                                if (h1 > 0) { // if hours (of hh:mm) is more than 0 then show how many hours you have, else only show mins below
                                    msg_3.push(layout_dict[layout[i]]);
                                    msg_3.push(h1 + ' h ' + m1 + ' m');
                                } else {
                                    msg_3.push(layout_dict[layout[i]]);
                                    msg_3.push(m1 + ' m ');
                                }
```

tovernaar123 replied:
makes sence what does not make sence

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                                    msg_3.push(m1 + ' m ');
+                                }
+                    
+                            } else {
+                                try {
+                                    msg_3.push(layout_dict[layout[i]]);
+                                    msg_3.push(ts(msg_2[layout[i]]));
+                                } catch (e) {
+                                    msg_3.push(layout_dict[layout[i]]);
+                                    msg_3.push(ts(0));
+                                }
+                            }
```
Again what are we doing here why are we calling ts with either msg_2[layout[i] or 0?
Also what does ts do? what does it return? why is it named ts? why is it an catch statement, is it going to error?
why will it error? so many questions

PHIDIAS0303 replied:
not all of the stuff have value inside.

tovernaar123 replied:
@PHIDIAS0303 
```suggestion
                            } else {
                                    msg_3.push(layout_dict[layout[i]]);
                                    if(msg_2 && layout[i]) {
                                           msg_3.push(ts(msg_2[layout[i]]));
                                    }else{
                                            msg_3.push(ts(0));
                                    }
                            }
```

tovernaar123 replied:
second why do we need to call ts on 0 wont this always be the same?

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                                }
+                            }
+                    
+                            msg_4.push([msg_3[0], msg_3[1]]);
```
what is this?

PHIDIAS0303 replied:
push title and value in array

aldldl replied:
```suggestion
                            msg_4.push([msg_3[0], msg_3[1]]); // push title and value in array
```

tovernaar123 replied:
titel of what, and which value

aldldl replied:
title of json, then value of it

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                    //Find "items" from datastore 
+                    const fs  = require('fs');
+                        // get raw data
+                    let rawdata = fs.readFileSync('/home/exp_admin/api_v2/persistent_storage.json');
```
like pd json i would like it to asycn and use await readFile, also a cache might be nice

aldldl replied:
I think I need help with this part @tovernaar123

tovernaar123 replied:
```js
                    const fs = require('fs');
                    const util = require('util');
                    const readFile = util.promisify(fs.readFile);
                    let rawdata = await readFile('/home/exp_admin/api_v2/persistent_storage.json');
```

tovernaar123 replied:
this is not needed and a cache is just storing the data in memory so it can be read quicker

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                        // Take raw data and change it into Json format, to make it simpler to format/lookup
+                    let mydata = JSON.parse(rawdata);
+                    //Use the key to look up the player data, for this current data key needs to be the player name to look up.
+                    let key1 = `${name}`; // in dataFile
+                    //check data is the parsed data, but only the PlayerData, and that that matches the key1 (name)
+                    let checkdata = mydata["PlayerData"][key1];
```
this can be  
let checkdata = mydata["PlayerData"][name];

aldldl replied:
```suggestion
                    //let key1 = `${name}`; // in dataFile // ->-> commented out till testing
                    //check data is the parsed data, but only the PlayerData, and that that matches the key1 (name)
                    let checkdata = mydata["PlayerData"][name];
```

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                            msg.channel.send('Name error: Name not found in datastore. Check the name or try again later.');console.log(`Name Not Found`); return;
+                        }
+                    //if it didnt stop based on the name not returining it will then filter out only the Statistics (removing prefrences like alt mode, join msg etc)
+                    finaldata = mydata["PlayerData"][key1]["Statistics"];
```
we can do 
finaldata = checkdata.Statistics

aldldl replied:
I tried that at one point, rather sure it didn't work for me... but I could have had other error at the same time as to why it didn't work.

tovernaar123 replied:
make sure its with a capital letter

aldldl replied:
ok

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                    item = finaldata;
+
+                    let channel = msg.channel;
+                    let result = profile(item);
```
what does profile do onther comment might be nice here stating what it returns.

aldldl replied:
@PHIDIAS0303 do you know?

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+
+                    let channel = msg.channel;
+                    let result = profile(item);
+                    let lookup = key1;
```
key1 is just the name you can just change this to name makes it more clear.

aldldl replied:
```suggestion
                    let lookup = name;
```

tovernaar123 replied:
why create onther varaible with the same value just change every place that uses lookup or key1 to name

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                    Embed.setColor("0x00ff00");
+
+                    for (let i = 0; i < result.length/2; i+=2) {
+                        try {
+                            Embed.addField(result[i][0], result[i][1], true);
+                        } catch (e) {
+                            Embed.addField(`** **`, `** **`, true);
+                        }
```
why will it fail?

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                    }
+
+                    //Send the Embeds, sent as 2 because depending on the length discord would error out if you sent them both.
+                    channel.send(Embed);
+                    channel.send(Embed2);
```
why not check if they are to big and only if they are send 2?

PHIDIAS0303 replied:
we are putting 0 in those cells, so it is fixed size

tovernaar123 replied:
okey ^

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+            } 
+            else 
+            {
+                msg.reply(`Something went wrong. See the logs because you should never run into this on this command`)
+                    .catch((err) => { internal_error(err); return })
+                console.log(`player data lookup by ${msg.author.username} but something happened, we dont know what, look at the logs that were just posted above.`);
+                return;
```
you shood run internal_error here

aldldl replied:
I think it does.
".catch((err) => { internal_error(err); return })"

tovernaar123 replied:
i am blind

tovernaar123 commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                    const Embed2 = new Discord.MessageEmbed();
+                    Embed2.setColor("0x00ff00");
+
+                    for (let i = 0; i < result.length/2; i+=2) {
+                        let j = i + result.length/2;
+                        try {
+                            Embed2.addField(result[j][0], result[j][1], true);
+                        } catch (e) {
+                            Embed2.addField(`** **`, `** **`, true);
+                        }
+
+                        try {
+                            Embed2.addField(result[j+1][0], result[j+1][1], true);
+                        } catch (e) {
+                            Embed2.addField(`** **`, `** **`, true);
+                        }
+                        
+                        Embed2.addField(`** **`, `** **`, true);
+                    }
```
what does this do?

tovernaar123 commented on infoBot.js:
```diff
@@ -130,9 +130,25 @@ client.on("message", async msg => {
-            console.log(`Unauthorized `);
-            msg.channel.send(`You do not have ${role.name} permission.`);
-            return;
+            if(command.name == `playerdata` && args[0] == msg.member.displayName || command.name == `playerdata2` && args[0] == msg.member.displayName || command.name == `playerdata2` && !args[0])
```
No you can do command parsing in infobot.js

aldldl replied:
what? this *is* infobot.js

tovernaar123 replied:
cant* this is not allowed this will make the code unusable with more commands

tovernaar123 replied:
all command must follow a standart way of 1 file and not changes to infobot.js cause that will create a big mess :(

aldldl replied:
Fair but we need to add a standard way to check the role then... we had one but we wanted to allow ALL users to check their own name on an otherwise restricted command. I guess we open it up to everyone and restrict it in the command. I just did not want it to show up in help at first.

aldldl replied:
Finished this for now, check out new code

tovernaar123 reviewed (changes requested):
Please look at my comments

tovernaar123 committed:
patch

tovernaar123 committed:
checking if this works

tovernaar123 committed:
fix

tovernaar123 committed:
a

tovernaar123 committed:
last try good fix

tovernaar123 committed:
typo

tovernaar123 committed:
typo2

aldldl committed:
Update infoBot.js

Added S8 to offline list.

aldldl reviewed (commented):
reviewed a bunch of stuff, made some suggested fixes in comments for final review, asked questions about questions I didn't understand.

aldldl commented on commands/playerdata.js:
```diff
@@ -0,0 +1,54 @@
+//player data storage access
+
```
```suggestion
            const fs  = require('fs');
```

aldldl commented on commands/playerdata.js:
```diff
@@ -0,0 +1,54 @@
+        }
+        if (name) 
+        {
+            const fs  = require('fs');
```
```suggestion

```

aldldl committed:
Update commands/playerdata.js

change to checkdata

aldldl committed:
Update commands/playerdata.js

change to finaldata

aldldl commented on commands/playerdata2.js:
```diff
@@ -0,0 +1,192 @@
+                            msg.channel.send('Name error: Name not found in datastore. Check the name or try again later.');console.log(`Name Not Found`); return;
+                        }
+                    //if it didnt stop based on the name not returining it will then filter out only the Statistics (removing prefrences like alt mode, join msg etc)
+                    finaldata = mydata["PlayerData"][key1]["Statistics"];
```
```suggestion
                    // OG Line -> finaldata = mydata["PlayerData"][key1]["Statistics"];
                    finaldata = checkdata.Statistics;
```

aldldl reviewed (commented):
fixed some more things

aldldl committed:
Apply suggestions from code review

Applied suggestions from Tov/ALo

Commit:
change from pd to pdj

Rename and change akas for the OG player data command. New command aka: ['pdj', 'userdatajson', 'oguserdata','pdjson','pdog'],

Commit:
change function location, rename command

Renamed the command from pd2 to pd, this is the new main command. Also moved the function based on last review.

Commit:
Try T1 error fix

t1 error fix with nan

Commit:
require discord

require discord at the top

Commit:
Finish aka updates

updated all akas to match properly

Commit:
update role allowed inside command rather than info.js

permissions inside the command.

Commit:
comment out exceptions for pd commands

Commit:
fixed async

fix async

Commit:
typo

Commit:
Added error message for self data

aldldl commented:
I am going to close this pull request and create a new one, the code has changed so much the old comments don't really make sense anymore. Will need help with the file cache thing, but all else should be much better.
